src/lexer.rs

```diff
@@ -34,6 +33,24 @@ pub enum TokenData<'t> {
     EOF,
 }
 
+impl<'t> fmt::Display for TokenData<'t> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use self::TokenData::*;
+
+        match self {
+            Keyword(keyword) => write!(f, "{}", keyword),
+            Operator(operator) => write!(f, "{}", operator),
+            Identifier(symbol) => write!(f, "identifier {}", symbol),
+            IntegerLiteral(symbol) => write!(f, "integer literal {}", symbol),
+            Comment(body) => write!(f, "/*{}*/", body),
+            UnclosedComment(body) => write!(f, "//{}", body),
```

Comment:
`UnclosedComment` is not `//`, but `/*`, without `*/`. Do we want to print this as `//`?

Reply:
If I understand correctly, UnclosedComments are not allowed, therefore this should be removed by @joshuabach patch that will also remove UnexpectedCharacter.

Reply:
`UnclosedComment` (now in `ErrorKind`) does not mean a `// comment`, but reaching end of file while parsing a `/* comment` ... So yes, it's an error now